src/OpenLoco/src/World/Station.cpp

```diff
@@ -1055,6 +1055,44 @@ namespace OpenLoco
         }
     }
 
+#pragma warning(push)
+#pragma warning(disable : 4858)
+    // 0x0048F482
+    void removeTileFromStation(const StationId stationId, const World::Pos3& pos, uint8_t rotation)
+    {
+        auto* station = StationManager::get(stationId);
+        auto findPos = pos;
+        findPos.z |= rotation;
+
+        // Bug mitigation: ensure stationTileSize does not exceed the actual array length
+        station->stationTileSize = std::clamp<uint16_t>(station->stationTileSize, 0, static_cast<uint16_t>(std::size(station->stationTiles)));
+
+        // Find tile to remove
+        auto foundTilePos = std::find(std::begin(station->stationTiles), std::end(station->stationTiles), findPos);
+        if (foundTilePos == std::end(station->stationTiles))
+            return;
+
+        // Remove tile by moving the remaining tiles over the one to remove
+        // NB: erasing is handled by StationManager::zeroUnused; not calling std::erase due to type mismatches
+        std::remove(std::begin(station->stationTiles), std::end(station->stationTiles), findPos);
```

Comment:
This is not ideal. I tried the following:
```suggestion
        auto pastLastTile = std::remove(std::begin(station->stationTiles), std::end(station->stationTiles), findPos);
        std::erase(pastLastTile, std::end(station->stationTiles));
```

But this yields the following type error:
```
/openloco/src/OpenLoco/src/World/Station.cpp: In function 'void OpenLoco::removeTileFromStation(StationId, const World::Pos3&, uint8_t)':
/openloco/src/OpenLoco/src/World/Station.cpp:1077:19: error: no matching function for call to 'erase(OpenLoco::Math::Vector::TVector3<short int, OpenLoco::World::WorldSpaceTag>*&, OpenLoco::Math::Vector::TVector3<short int, OpenLoco::World::WorldSpaceTag>*)'
 1077 |         std::erase(pastLastTile, std::end(station->stationTiles));
      |         ~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/i686-w64-mingw32/sys-root/mingw/include/c++/functional:64,
                 from /openloco/src/Core/include/OpenLoco/Core/BitSet.hpp:8,
                 from /openloco/src/OpenLoco/src/World/Company.h:7,
                 from /openloco/src/OpenLoco/src/World/CompanyManager.h:3,
                 from /openloco/src/OpenLoco/src/World/Station.cpp:2:
```

Reply:
That is correct its not a variable sized container you can't erase it. All we need to do is remove which shifts everything left. Tbh. its implementation defined what goes in to the end element but i think we can expect it will be what we want.

Reply:
That sounds like a good idea to work around the warning. Eventually `zeroUnused` will zero out the memory, anyway. I'll try this.